src/components/TimeseriesChartExport/TimeseriesChartExport.tsx

```diff
@@ -0,0 +1,416 @@
+
+        if (onSuccess) {
+          onSuccess();
+        }
+
+        downloadCSV(csvString);
+      } catch (e) {
+        setLoading(false);
+
+        if (onError) {
+          onError(e);
+        }
+      }
+    });
+  };
+
+  useEffect(() => {
+    if (cellLimit > CELL_LIMIT) {
+      setLimit(CELL_LIMIT);
+    }
+  }, [cellLimit]);
+
+  useEffect(() => {
+    const { range, granularity: granularityString } = form.getFieldsValue();
+    const start = range[0] ? +range[0] : 0;
+    const end = range[1] ? +range[1] : 0;
+    const granularityValue = getGranularityInMS(granularityString);
+    const seriesNumber = timeseries.length;
+
+    setLimitHit(
+      isGreaterThenLimit(limit, start, end, granularityValue, seriesNumber)
+    );
+  }, []);
+
+  useEffect(() => {
+    if (!visible || !timeseries.length) {
+      return;
+    }
+
+    const fetchTimeseries = retrieveTimeseries || context!.timeseries.retrieve;
+
+    setLoading(true);
+
+    fetchTimeseries(timeseries)
+      .then(t => {
+        setSeries(t);
+        setLoading(false);
+      })
+      .catch(error => {
+        setLoading(false);
+
+        if (onError) {
+          onError(error);
+        }
+      });
+  }, [visible]);
+
+  return (
+    <Modal
+      visible={visible}
+      title={title}
+      width={modalWidth}
+      footer={null}
+      onCancel={hideModal}
+    >
+      <Form data-test-id="form" onSubmit={onSubmit}>
+        <Form.Item {...formItemLayout} label={labelRange}>
+          {getFieldDecorator('range', {
+            initialValue: [moment(defaultRange[0]), moment(defaultRange[1])],
+            rules: [{ required: true }],
+          })(
+            <RangePicker
+              onChange={checkLimitOnRangeChange}
+              showTime={{ format: 'HH:mm' }}
+              format="YYYY-MM-DD HH:mm"
+              style={{ width: '100%' }}
+            />
+          )}
+        </Form.Item>
+        <Form.Item
+          {...formItemLayout}
+          label={labelGranularity}
+          help={labelGranularityHelp}
+        >
+          {getFieldDecorator('granularity', {
+            rules: [{ required: true }],
+            initialValue: granularity,
+          })(
+            <Input
+              data-test-id="granularity"
+              onChange={checkLimitOnGranularityChange}
+            />
+          )}
+        </Form.Item>
+        <Form.Item
+          {...formItemLayout}
+          label={formatTimestamp}
+          help={formatTimestampHelp}
+        >
+          {getFieldDecorator('readableDate', {
+            valuePropName: 'checked',
+          })(<Checkbox />)}
+        </Form.Item>
+        <Form.Item
+          {...formItemLayout}
+          label={delimiterLabel}
+          help={delimiterHelp}
+        >
+          {getFieldDecorator('delimiter', {
+            initialValue: ',',
+            rules: [{ required: true }],
+          })(
+            <Radio.Group>
+              {Object.values(Delimiters).map(delimiter => (
```

Comment:
maybe `Object.keys(Delimiters).map` better?  then we just display them as `Comma|Tab|Semicolon`, not with symbols + we don't need that logic `{delimiter === Delimiters.Tab ? 'tab' : delimiter}` anymore
(I know it's different from opint)

Reply:
What about translations? Do we need to translate it here? With signs, we don't need to think about it

Reply:
yeah, good point